reframe/frontend/statistics.py

```diff
@@ -1,89 +1,98 @@
+from itertools import chain
+
 import reframe.core.debug as debug
+import reframe.core.exceptions as exc
 
 
 class TestStats:
     """Stores test case statistics."""
 
-    def __init__(self, tasks=[]):
-        if not hasattr(tasks, '__iter__'):
-            raise TypeError('expected an iterable')
-
-        # Store test cases per partition internally
-        self._tasks_bypart = {}
-        for t in tasks:
-            partition = t.check.current_partition
-            partname = partition.fullname if partition else 'None'
-            tclist = self._tasks_bypart.setdefault(partname, [])
-            tclist.append(t)
+    def __init__(self):
+        self._tasks = []
+        self._try_num = -1
```

Comment:
This variable should hold the current run (name it also accordingly). Just that. It will be used by the `add_task` to append a task to the right list.

Reply:
I now initialize with `self._try_num = 0`. Done.

Reply:
OK, `_current_run` seems perfect to me. Done.